### DISCUSSION TITLE
chat

### DISCUSSION THREAD

**Discussion by nobeeakon** [OSGeo/grass#2505] [General]
Hey guys, just joining the project so most probaly this has already been discussed at some point ...

I think it may be useful to have a slack, discord or some other communication channel to, well, ease the communication. Think those are useful to ask quick things, or share some ideas or request feedback, and more agile than other ways. I mean, it is not a replacement of things like the mailing lists and other communication ways you guys have and maybe it is just that I'm new here (maybe more useful for new comers? haha) , but think it would be useful. 

Some  projects have them, some don't so guess that as everything it comes with some downsides...

**Comment by neteler** [comment]
At least there is the "good old" IRC:
https://grasswiki.osgeo.org/wiki/How_to_participate_in_IRC_communication

but other media may appear more appealing.

**Reply by nobeeakon** [reply]
guess that the thing is if using those other tools would  benefit the project and the community (devs + users).  It is, as far as I can see, an open question (for example see this discussion [drupal guys had](https://groups.drupal.org/node/535264) about migrating a community from IRC to an alternative ). As noted in [this discussion](https://forum.strapi.io/t/should-be-move-the-community-irc-from-slack-to-discord/3603/3), other projects (even massive ones like  Next.js, Typescript, Vue, Free Code Camp, and many others...) are already using Slack or Discord.

**Comment by neteler** [comment]
BTW: I like this solution here:

https://miktex.org/
--> Community

![image](https://user-images.githubusercontent.com/1295172/182449944-5e1e2d92-b872-42d4-9689-0ba94230a6af.png)

**Reply by nobeeakon** [reply]
which brings back to [issue 245](https://github.com/OSGeo/grass-website/issues/245) -and sure other discussions I'm not aware of.  Yet, I think those are alternative means of asking for help. It's just that sometimes (during dev or even as normal user) it is good to have a more direct channel to communicate. For example QGIS even has a Telegram chat (don't know   if that's a  good approach or not but people use it) maybe not the most documentable/trackable thing, but it just makes it easy for people to ask I guess

**Reply by veroandreo** [reply]
This looks very nice and appealing, but it seems to me it's something useful to browse only the n latest items in the archives of questions or emails, i.e., not for direct chatting. 

I don't know if the grass IRC channel is still used or not (I think I never logged in there), so, difficult to say if posting a question there will get  you any answer. Indeed, I have thought of creating a Telegram group (mostly because it's the only chat I use while in the laptop), but it would be nice to understand if others find it a useful channel. It does allow to share files, pics, links, audio, code and you can search easily. In any case, it would be nice to understand what chat tool is mostly used/preferred...

**Reply by nobeeakon** [reply]
@veroandreo , at least in my own view the most widely used are Slack and Discord...
Here a small (personal)  comparison, hope it is useful. All of them can be used to share files, pics, video, etc.

- Telegram
   -  Pros: no cost, widely used,  
   - Cons: mobile oriented (although it can be used in the web or desktop)  it is mainly a Whatsapp competitor, more on the side of 1:1 communication than on the side of communities.
   
- Slack
   -  Pros: no cost (up to 10,000 messages). Widely used. Used by companies and communities. Allows creating dedicated topics chat rooms within the same general slack place, allows private chats, allows creating and answering in threads, allows 1:1 communication. 
   - Cons: message limit means eventually you loose previous conversations unless you pay. 
   (screenshot in case you haven't used it)
![imagen](https://user-images.githubusercontent.com/12853324/183463312-d63c0593-5b7a-42c3-9d87-f3e152044633.png)

- Discord
   -  Pros: no cost. No message limit. Widely used. Preferred by communities, gammers, streamers, etc. Allows creating dedicated topics chat rooms within the same general discord place, allows private chats,  allows 1:1 communication. 
   - Cons: No threads. Only pay for additional features if you want.
   (screenshot in case you haven't used it)
![imagen](https://user-images.githubusercontent.com/12853324/183463856-674052d4-7550-4143-8984-fca099b22d43.png)

- IRC
   - Pros: no cost. Not owned by a company  or external provider  (as opposed to Slack, Telegram or Discord). No need to create an account (as opposed to Slack, Telegram or Discord). 
   - Cons: from what I can see it may not be the most preferred way to communicate (even between grass devs)

In my opinion, Discord seems a better alternative than the others. Please have a look at these two interesting discussions [here](https://groups.drupal.org/node/535264), and [here](https://forum.strapi.io/t/should-be-move-the-community-irc-from-slack-to-discord/3603).  As noted in one of the comments on the second one, some massive projects like Vue, Next.js, Typescript, etc. (for context, these are huge web dev projects, some of which are OSS) are already using Discord to let the community of users communicate. 


Hope this helps.

**Reply by petrasovaa** [reply]
Some communities use [Gitter](https://gitter.im/).  I personally prefer more asynchronous communication, but I am open to trying something new.

**Reply by lbartoletti** [reply]
> IRC
> Cons: from what I can see it may not be the most preferred way to communicate (even between grass devs)

Grass have an IRC channel on [libera.chat]https://wiki.osgeo.org/wiki/IRC#Libera) managed by the [OSGeo IRC team](https://wiki.osgeo.org/wiki/IRC) (I'm a member of this team).
IRC can be "old school", but you can chat on with a modern protocol thanks to the matrix bridge: https://matrix.to/#/#grass:libera.chat

**Comment by tmszi** [comment]
From my own perspective as developer and user I prefer [Free software ](https://en.wikipedia.org/wiki/Free_software)('free' as in '[free speech](https://en.wikipedia.org/wiki/Freedom_of_speech)', not as in 'free beer') collaborative/chat tool ([KISS](https://en.wikipedia.org/wiki/KISS_principle)/[UNIX](https://en.wikipedia.org/wiki/Unix_philosophy) principles) over 'popular' proprietary collaborative/chat tool (which track and collect user data) for many reasons of course.

From the world of Free software, I bring to your attention the collaboration/chat software [Mattermost](https://mattermost.com/).

**Reply by nobeeakon** [reply]
hey @tmszi , I agree with the Free software principles you mention. Never used Mattermost, looks really nice, thanks.  The only thing I'd worry is that 'popularity' may not be a secondary thing when thinking in community engagement and so, specially if the community is not only developers but also users. May not be a relevant example here but, thinking in something like what happens with LibreOffice vs MS Office, people just prefer to use MS Office (for whatever reason) and, at least in my experience, it is not so easy to convince them to even try LibreOffice.

**Comment by veroandreo** [comment]
Hello everyone! Greetings from the FOSS4G 2022 code sprint in Florence! We have created a gitter channel for chatting: https://gitter.im/grassgis/community

You are all welcome to join!
